text/0024-track-validator-set-when-synchronizing.org

```diff
@@ -0,0 +1,142 @@
+#+STARTUP: inlineimages
+
+* Joining Node Validator Set Tracking When Synchronizing
+:PROPERTIES:
+:CUSTOM_ID: joining-node-validator-set-tracking-when-synchronizing
+:END:
+
+** Summary
+:PROPERTIES:
+:CUSTOM_ID: summary
+:END:
+
+CEP PR: [[https://github.com/casperlabs/ceps/pull/0024][casperlabs/ceps#0024]]
+
+A node joining the network needs a secure means of tracking the
+validator set as it processes blocks. To achieve this, the validators
+for the next era are to be provided in each switch block.
+
+This is required for [[https://github.com/casperlabs/ceps/pull/1234][casperlabs/ceps#1234 (placeholder link)]] and
+[[https://github.com/casperlabs/ceps/pull/9999][casperlabs/ceps#9999 (placeholder link)]].
+
+** Motivation
+:PROPERTIES:
+:CUSTOM_ID: motivation
+:END:
+
+As a node gets blocks and becomes up-to-date, it will process blocks
+in ascending eras. If the node has secure knowledge of the validators
+in that era it can use /finality signatures/ to determine the
+authenticity of block hashes. A procedure for tracking the validators
```

Comment:
The authenticity of block hashes or blocks? Is that a mental shortcut?

Reply:
I believe I mean block hashes.  Maybe a mental shortcut...?  Not sure what you mean.

Reply:
The way I was seeing it was that finality signatures are used to determine an authenticity of a block we're downloading. I guess you can say _block hash_ as well since it uniquely identifies a block. Although, for the purposes of the joining process a node wants whole blocks, not only block hashes. That's what I called a mental shortcut.